integration-tests/src/tests/nearcore/sync_state_nodes.rs

```diff
@@ -530,3 +535,141 @@ fn sync_state_dump() {
+fn test_dump_epoch_missing_chunk_in_last_block() {
+    heavy_test(|| {
+        init_test_logger();
+        let epoch_length = 10;
+
+        for num_last_chunks_missing in 0..3 {
+            assert!(num_last_chunks_missing < epoch_length);
+            let mut genesis =
+                Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
+            genesis.config.epoch_length = epoch_length;
+            let chain_genesis = ChainGenesis::new(&genesis);
+
+            let num_clients = 2;
+            let env_objects =
+                (0..num_clients)
+                    .map(|_| {
+                        let tmp_dir = tempfile::tempdir().unwrap();
+                        // Use default StoreConfig rather than NodeStorage::test_opener so we’re using the
+                        // same configuration as in production.
+                        let store =
+                            NodeStorage::opener(&tmp_dir.path(), false, &Default::default(), None)
+                                .open()
+                                .unwrap()
+                                .get_hot_store();
+                        let epoch_manager =
+                            EpochManager::new_arc_handle(store.clone(), &genesis.config);
+                        let runtime = NightshadeRuntime::test(
+                            tmp_dir.path(),
+                            store.clone(),
+                            &genesis,
+                            epoch_manager.clone(),
+                        ) as Arc<dyn RuntimeAdapter>;
+                        (tmp_dir, store, epoch_manager, runtime)
+                    })
+                    .collect::<Vec<(
+                        tempfile::TempDir,
+                        Store,
+                        Arc<EpochManagerHandle>,
+                        Arc<dyn RuntimeAdapter>,
+                    )>>();
+
+            let stores = env_objects.iter().map(|x| x.1.clone()).collect();
+            let epoch_managers = env_objects.iter().map(|x| x.2.clone()).collect();
+            let runtimes = env_objects.iter().map(|x| x.3.clone()).collect();
+
+            let mut env = TestEnv::builder(chain_genesis)
+                .clients_count(num_clients)
+                .stores(stores)
+                .epoch_managers(epoch_managers)
+                .runtimes(runtimes)
+                .use_state_snapshots()
+                .build();
+
+            let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap();
+            let mut blocks = vec![genesis_block.clone()];
+            let signer =
+                InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+            let target_height = epoch_length + 1;
+            for i in 1..=target_height {
+                let block = env.clients[0].produce_block(i).unwrap().unwrap();
+                blocks.push(block.clone());
+                if (i % epoch_length) != 0
```

Comment:
doesn't this mean the last block of an epoch won't be missing any chunks? I think the last block of an epoch missing chunks is exactly what we want to test for?